core/lib/snapshots_applier/src/lib.rs

```diff
@@ -109,16 +74,27 @@ impl From<RpcError> for SnapshotsApplierError {
     }
 }
 
+/// Main node API used by the [`SnapshotsApplier`].
 #[async_trait]
 pub trait SnapshotsApplierMainNodeClient: fmt::Debug + Send + Sync {
     async fn fetch_l2_block(&self, number: MiniblockNumber) -> Result<Option<SyncBlock>, RpcError>;
 
     async fn fetch_newest_snapshot(&self) -> Result<Option<SnapshotHeader>, RpcError>;
 }
-impl<'a, 'b> SnapshotsApplier<'a, 'b> {
-    pub async fn prepare_applied_snapshot_status(
+
+/// Applying application-level storage snapshots to the Postgres storage.
+#[derive(Debug)]
+pub struct SnapshotsApplier<'a> {
+    connection_pool: &'a ConnectionPool,
+    blob_store: &'a dyn ObjectStore,
```

Comment:
Lifetimes newbie question: I don't think I see, why it's better to have a single lifetime here

Reply:
It's a good question. Less lifetime params makes code easier to read / maintain, and in this case one lifetime is sufficient: because we use lifetimes for reference types in the same struct, their lifetimes are guaranteed to intersect (otherwise, we wouldn't be able to construct a stuct). This means that both refs can use this common lifetime `'a`; their original lifetimes will be automatically coerced to `'a` (in type theory terms, reference types are [covariant](https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science)) w.r.t. the lifetime).

As a counter-example, DAL types cannot use a single lifetime because they wrap `&'a mut StorageProcessor<'c>` which is invariant over `'c` (i.e., `'c` cannot be replaced with `'a`) because `&mut T` references are invariant over `T`.